File: src/game/hardwareinput/Command.java
package game.hardwareinput;

import game.entity.models.Direction;

public enum Command {
    MOVE_UP,
    MOVE_DOWN,
    MOVE_RIGHT,
    MOVE_LEFT,
    PLACE_BOMB,
    PAUSE,
    INTERACT;

    public Direction commandToDirection(){
        return switch (this) {
            case MOVE_UP -> Direction.UP;
            case MOVE_DOWN -> Direction.DOWN;
            case MOVE_LEFT -> Direction.LEFT;
            case MOVE_RIGHT -> Direction.RIGHT;
            default -> null;
        };
    }
}

File: src/game/hardwareinput/Command.kt
package game.hardwareinput;

import game.entity.models.Direction;

public enum Command {
    MOVE_UP,
    MOVE_DOWN,
    MOVE_RIGHT,
    MOVE_LEFT,
    ATTACK,
    PAUSE,
    INTERACT;

    public Direction commandToDirection() {
        switch (this) {
            case MOVE_UP:
                return Direction.UP;
            case MOVE_DOWN:
                return Direction.DOWN;
            case MOVE_LEFT:
                return Direction.LEFT;
            case MOVE_RIGHT:
                return Direction.RIGHT;
        }
        return null;
    }
}
